linux/gatt/client.go

```diff
@@ -212,10 +212,30 @@ func (p *Client) ReadCharacteristic(c *ble.Characteristic) ([]byte, error) {
 }
 
 // ReadLongCharacteristic reads a characteristic value which is longer than the MTU. [Vol 3, Part G, 4.8.3]
-func (p *Client) ReadLongCharacteristic(c *ble.Characteristic) ([]byte, error) {
+func (p *Client) ReadLongCharacteristic(c *ble.Characteristic) ([]byte,error) {
 	p.Lock()
 	defer p.Unlock()
-	return nil, nil
+
+	buffer := make([]byte,512)
+	size := 0
+	part := 0
+
+	read, err := p.ac.Read(c.ValueHandle)
+	for {
+		if err != nil {
```

Comment:
I think it is cleaner and more expressive to handle the error where it appears, even if this means we have to duplicate the error check for `p.ac.Read()` and `p.ac.ReadBlob()`